File: goblin-embedded/src/main/java/org/goblinframework/embedded/core/provider/JdkEmbeddedServerFactory.java
package org.goblinframework.embedded.core.provider;

import org.goblinframework.embedded.core.setting.ServerSetting;
import org.goblinframework.embedded.server.EmbeddedServer;
import org.goblinframework.embedded.server.EmbeddedServerFactory;
import org.goblinframework.embedded.server.EmbeddedServerMode;
import org.jetbrains.annotations.NotNull;

public class JdkEmbeddedServerFactory implements EmbeddedServerFactory {

  @NotNull
  @Override
  public EmbeddedServerMode mode() {
    return EmbeddedServerMode.JDK;
  }

  @NotNull
  @Override
  public EmbeddedServer createEmbeddedServer(@NotNull ServerSetting setting) {
    return new JdkEmbeddedServer(setting);
  }
}

File: goblin-embedded/src/main/java/org/goblinframework/embedded/java/JdkEmbeddedServerFactory.kt
package org.goblinframework.embedded.java

import org.goblinframework.api.annotation.Singleton
import org.goblinframework.embedded.core.setting.ServerSetting
import org.goblinframework.embedded.server.EmbeddedServer
import org.goblinframework.embedded.server.EmbeddedServerFactory
import org.goblinframework.embedded.server.EmbeddedServerMode

@Singleton
class JdkEmbeddedServerFactory private constructor() : EmbeddedServerFactory {

  companion object {
    @JvmField val INSTANCE = JdkEmbeddedServerFactory()
  }

  override fun mode(): EmbeddedServerMode {
    return EmbeddedServerMode.JAVA
  }

  override fun createEmbeddedServer(setting: ServerSetting): EmbeddedServer {
    return JdkEmbeddedServer(setting)
  }
}